core/src/main/java/io/grpc/internal/InternalSubchannel.java

```diff
@@ -555,10 +492,10 @@ private static void checkListHasNoNulls(List<?> list, String msg) {
     @Override
     public void transportReady() {
       channelLogger.log(ChannelLogLevel.INFO, "READY");
-      Status savedShutdownReason;
-      try {
-        synchronized (lock) {
-          savedShutdownReason = shutdownReason;
+      syncContext.execute(new Runnable() {
+        @Override
+        public void run() {
+          Status savedShutdownReason = shutdownReason;
```

Comment:
We can safely use `shutdownReason` directly within this scope.

Reply:
yes thanks.